### ISSUE TITLE
clang-tidy marks as error compile flag -static-libasan [clang-diagnostic-error] 

### ISSUE THREAD

**Issue by h1laryz** [llvm/llvm-project#82314]
Hi, I am using gcc/g++ as compilers in my projects, but also I want to use .clang-tidy as static analyzer.
While building I am getting an error

```
error: unknown argument '-static-libasan'; did you mean '-static-libsan'? [clang-diagnostic-error]
14068 warnings and 1 error generated.
```

How can I disable this warning in **_.clang-tidy_**?
Here is my **_.clang-tidy_** file:
```
Checks: >
bugprone-,
cppcoreguidelines-,
modernize-,
performance-,
-cppcoreguidelines-avoid-magic-numbers,
-cppcoreguidelines-pro-bounds-array-to-pointer-decay,
-cppcoreguidelines-pro-type-reinterpret-cast,
-cppcoreguidelines-pro-type-vararg,
-modernize-use-trailing-return-type,
readability-magic-numbers,
google-runtime-int,
misc-,
clang-analyzer-,

WarningsAsErrors: '*'
HeaderFilterRegex: 'a^'
FormatStyle: file
```

this from CMakeLists.txt getting an error during compilation. Because clang-tidy says it is an error.

```
set_target_properties(
${PROJECT_NAME} PROPERTIES
CXX_CLANG_TIDY "${CLANG_TIDY_EXECUTABLE}"
)
```

Clangd I am using for vim/vscode and it also says that this -static-libasan flag is unknown, but actually this flag comes from a third_party library.

![изображение](https://github.com/llvm/llvm-project/assets/71406854/eee4fd02-ed71-49e3-8382-446434b2b1ec)

How can I disable this error in **_.clang-tidy_** checks?
I tried to add something like this:
```
-clang-diagnostic-unknown-argument,
-no-unknown-warning-option,
-unknown-argument,
```

but it didn't help at all.

**Comment by PiotrZSL** [comment]
What version of clang-tidy ?
And this is probably due to WarningsAsErrors

**Comment by HighCommander4** [comment]
The issue does not require WarningsAsErrors. A minimal set of files reproducing the issue is:

compile_commands.json:
```json
[
    {
        "arguments": [
            "g++",
            "-c",
            "-static-libasan",
            "test.cpp"
        ],
        "directory": "/path/to/project",
        "file": "test.cpp"
    }
]
```

test.cpp:
```c++
int main() {}
```

Then running `clang-tidy test.cpp` gives:

```console
$ clang-tidy test.cpp
1 error generated.
Error while processing /path/to/project/test.cpp.
error: unknown argument '-static-libasan'; did you mean '-static-libsan'? [clang-diagnostic-error]
Found compiler error(s).
```

**Comment by h1laryz** [comment]
> What version of clang-tidy ? And this is probably due to WarningsAsErrors
```
h1lary@h1larypc:~$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 10.0.0

  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: skylake

h1lary@h1larypc:~$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

h1lary@h1larypc:~$ g++ --version
g++ (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

h1lary@h1larypc:~$ cmake --version
cmake version 3.28.3
CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

Yes, actually I want to use warnings as errors, but I need to fix this warning because of using gcc/g++ compilers

**Comment by h1laryz** [comment]
Actually, I am getting this compile flag from this library. Here is their .clang-tidy:
https://github.com/userver-framework/userver/blob/develop/.clang-tidy
Maybe, you can find a solution from it

**Comment by PiotrZSL** [comment]
Clang does not support GCC switches, so simply remove them command line and from compile_commands.json, same goes for unknown warnings
```
clang++ -static-libasan
clang++: error: unknown argument '-static-libasan'; did you mean '-static-libsan'?
clang++: error: no input files
```

**Comment by HighCommander4** [comment]
> simply remove them [...] from compile_commands.json

Technically, it's fair to say that users should just make sure their `compile_commands.json` contains 100% clang-compatible commands, and leave it at that.

However, the practical reality is that (1) many users use compilers other than clang to build their code, and (2) tools for generating `compile_commands.json` generally produce the same commands that were used during the build.

So, I think some mechanism for clang-based tools to accept commands which are not 100% clang-compatible would be helpful.

In clangd, what we've done is allow users to post-process their compile commands using a clangd config file, e.g. in this case you can create a `.clangd` file containing:

```yaml
CompileFlags:
  Remove: [-static-libasan]
```

Perhaps it would make sense for clang-tidy to consider doing something similar?

**Comment by vbvictor** [comment]
The feature was added in https://github.com/llvm/llvm-project/pull/164344
